controllers/argocd/notifications.go

```diff
@@ -520,7 +549,163 @@ func (r *ReconcileArgoCD) reconcileNotificationsSecret(cr *argoproj.ArgoCD) erro
+				Namespace: sourceNamespace,
+				Labels:    argoutil.LabelsForCluster(cr),
+			},
+			Rules: policyRuleForNotificationsController(),
+		}
+		err = r.reconcileSourceNamespaceRole(role, cr)
+		if err != nil {
+			reconciliationErrors = append(reconciliationErrors, err)
+		}
+
+		roleBinding := rbacv1.RoleBinding{
+			ObjectMeta: v1.ObjectMeta{
+				Name:        getResourceNameForNotificationsSourceNamespaces(cr),
+				Labels:      argoutil.LabelsForCluster(cr),
+				Annotations: argoutil.AnnotationsForCluster(cr),
+				Namespace:   sourceNamespace,
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "Role",
+				Name:     getResourceNameForNotificationsSourceNamespaces(cr),
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      rbacv1.ServiceAccountKind,
+					Name:      getServiceAccountName(cr.Name, "notifications-controller"),
+					Namespace: cr.Namespace,
+				},
+			},
+		}
+		err = r.reconcileSourceNamespaceRoleBinding(roleBinding, cr)
+		if err != nil {
+			reconciliationErrors = append(reconciliationErrors, err)
+		}
+
+		// ensure NotificationsConfiguration CR exists in the source namespace
+		if err := r.reconcileSourceNamespaceNotificationsConfigurationCR(cr, sourceNamespace); err != nil {
+			reconciliationErrors = append(reconciliationErrors, err)
+		}
+
+		// notifications permissions for argocd server in source namespaces are handled by apps-in-any-ns code
+
+		if _, ok := r.ManagedNotificationsSourceNamespaces[sourceNamespace]; !ok {
+			if r.ManagedNotificationsSourceNamespaces == nil {
+				r.ManagedNotificationsSourceNamespaces = make(map[string]string)
+			}
+			r.ManagedNotificationsSourceNamespaces[sourceNamespace] = ""
+		}
+	}
+
+	return amerr.NewAggregate(reconciliationErrors)
+}
+
+// reconcileSourceNamespaceNotificationsConfigurationCR ensures a NotificationsConfiguration CR exists in the given source namespace.
+func (r *ReconcileArgoCD) reconcileSourceNamespaceNotificationsConfigurationCR(cr *argoproj.ArgoCD, sourceNamespace string) error {
+	if !isNotificationsEnabled(cr) {
+		return nil
+	}
+
+	// Check if NotificationsConfiguration exists in source namespace
+	sourceNotifCfg := &v1alpha1.NotificationsConfiguration{}
+	err := argoutil.FetchObject(r.Client, sourceNamespace, DefaultNotificationsConfigurationInstanceName, sourceNotifCfg)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to get the NotificationsConfiguration from source namespace %s : %s", sourceNamespace, err)
+		}
+		// Not found in source namespace, create it with propagated/default spec
+		newCfg := &v1alpha1.NotificationsConfiguration{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      DefaultNotificationsConfigurationInstanceName,
+				Namespace: sourceNamespace,
+			},
+			Spec: v1alpha1.NotificationsConfigurationSpec{
+				Context:   getDefaultNotificationsContext(),
+				Triggers:  getDefaultNotificationsTriggers(),
+				Templates: getDefaultNotificationsTemplates(),
```

Comment:
This will still populate the default notifications configs. Can you try assigning a empty map?
```suggestion
				Context:   make(map[string]string),
				Triggers:  make(map[string]string),
				Templates: make(map[string]string),
```